backend/src/LeanCode.AppRating/IUserIdExtractor.cs

```diff
@@ -4,5 +4,5 @@ namespace LeanCode.AppRating;
 
 public interface IUserIdExtractor<TUserId>
 {
-    public TUserId Extract(HttpContext httpContext);
+    public bool TryExtract(HttpContext httpContext, out TUserId? userId);
```

Comment:
I would suggest having both, one that throws, and one that doesn't. The one can be implemented (w/ default interface implementation) using the other.